modules/docs/arrow-docs/docs/docs/arrow/typeclasses/traverse/README.md

```diff
@@ -11,16 +11,366 @@ redirect_from:
+
+If we fix `D` to be some sort of dependency or configuration, we can use the `Reader` applicative in our `traverse`.
+
+```kotlin:ank
+import arrow.mtl.Reader
+
+interface Context
+interface Topic
+interface Result
+
+typealias Job<A> = Reader<Context, A>
+
+fun processTopic(t: Topic): Job<Result> = TODO()
+```
+
+We can imagine we have a data pipeline that processes a bunch of data, each piece of data being categorized by a topic. Given a specific topic, we produce a `Job` that processes that topic. (Note that since a `Job` is just a `Reader`/`Kleisli`, one could write many small `Jobs` and compose them together into one `Job` that is used/returned by `processTopic`.)
+
+Corresponding to our bunches of data are bunches of topics, a `List<Topic>` if you will. Since Reader has an `Applicative` instance, we can `traverse` over this list with `processTopic`.
+
+```kotlin:ank
+import arrow.core.Id
+import arrow.core.ForId
+import arrow.core.ListK
+import arrow.core.extensions.id.applicative.applicative
+import arrow.mtl.KleisliPartialOf
+import arrow.mtl.extensions.kleisli.applicative.applicative
+import arrow.mtl.fix
+
+fun processTopics(topics: ListK<Topic>): Job<ListK<Result>> =
+  topics.traverse<KleisliPartialOf<ForId, Context>, Result>(Job.applicative<ForId, Context>(Id.applicative())) {
+    processTopic(it)
+  }.fix()
+```
+
+Note the nice return type - `Job<List<Result>>`. We now have one aggregate `Job` that when run, will go through each topic and run the topic-specific job, collecting results as it goes. We say "when run" because a `Job` is some function that requires a Context before producing the value we want.
+
+One example of a "context" can be found in the [Spark](http://spark.apache.org/) project. In Spark, information needed to run a Spark job (where the master node is, memory allocated, etc.) resides in a `SparkContext`. Going back to the above example, we can see how one may define topic-specific Spark jobs `(type Job<A> = Reader<SparkContext, A>)` and then run several Spark jobs on a collection of topics via `traverse`. We then get back a `Job<List<Result>>`, which is equivalent to (`SparkContext) -> List<Result>`. When finally passed a `SparkContext`, we can run the job and get our results back.
+
+Moreover, the fact that our aggregate job is not tied to any specific `SparkContext` allows us to pass in a `SparkContext` pointing to a production cluster, or (using the exact same job) pass in a test `SparkContext` that just runs locally across threads. This makes testing our large job nice and easy.
+
+Finally, this encoding ensures that all the jobs for each topic run on the exact same cluster. At no point do we manually pass in or thread a `SparkContext` through - that is taken care for us by the (applicative) effect of `Reader` and therefore by `traverse`.
+
+### Sequencing
```

Comment:
TBH given the number of requests we get I'd start with sequencing, and then explaining that traverse is a specialization.

Reply:
Exactly. The most usual problem is the one about "reversing the types" which is what sequence resolves. You can start with sequence, then in the end unveil its equivalent to traverse with identity, then explain Traverse and how it's even more powerful being able to use other transformation functions.